src/openApi/openApiWrapper.ts

```diff
@@ -48,36 +57,42 @@ export default class OpenApiWrapper {
             statusCode: response.statusCode,
           });
 
+          const { correlationId } = this;
           this.clearContext();
 
+          response.headers = Object.assign(response.headers ?? {}, {
```

Comment:
I don't follow. The type here is `ApiResponse`, why the `.withCorrelationId(correlationId);` mutator cannot be used?

I know you were saying that there are no types for middleware, but I see them?
```
responseMiddleware: (request: ApiRequest<unknown>, response: ApiResponse)
```

Reply:
Because when I use `lambda-essentials-ts` in my service, I can do this (related to the other comment where I was proposing to type the hooks to avoid this issue)
![image](https://user-images.githubusercontent.com/11741490/106924800-bee85680-670f-11eb-8cc2-cb70934e0d92.png)

Reply:
Ok, now everything clicks! Wow, then either implement the types or change the signature of `responseMiddleware` to reflect that can happen, e.g. `ApiResponse | unknown` 🤦

Reply:
Removed this shady logic after adding OpenApiModel